back-end/app.js

```diff
@@ -14,11 +17,30 @@ app.use(express.urlencoded({ extended: false }));
 app.use(cookieParser());
 app.use(express.static(path.join(__dirname, 'public')));
 
+app.use(
+    session({
+        secret: "your_secret_key",
+        resave: false,
+        saveUninitialized: false,
+    })
+);
+
+app.use(passport.initialize());
+app.use(passport.session());
+
 app.use('/', indexRouter);
 app.use('/users', usersRouter);
+app.use("/auth", authRoutes);
+
+app.use(errorHandler)
 
 app.listen(4000, () => {
   console.log("Server is running on localhost:4000, you better catch it!");
 });
 
+function errorHandler (err, req, res, next) {
+  //simple error response
+  return res.status(401).send(err.message)
```

Comment:
Better if this one sends the expected error pattern and only sets the status code if we don't ahead of time.
```suggestion
  return res.status(res.statusCode !== 200 ? res.statusCode : 500).json({ message: err.message })
```